nodedb.go

```diff
@@ -211,6 +219,10 @@ func (ndb *nodeDB) SaveFastNode(node *FastNode) error {
 }
 
 func (ndb *nodeDB) setStorageVersion(newVersion string) error {
+	if newVersion >= fastStorageVersionValue {
+		newVersion = newVersion + fastStorageVersionDelimiter + strconv.Itoa(int(ndb.latestVersion))
```

Comment:
won't we be infinitely appending to newVersion like this, if newVersion gets parsed from the serialized value later?

Reply:
I don't think any of the tests test repeated calls here

Reply:
You are right, good catch.

I refactored this method completely and renamed it to `setFastStorageVersionToBatch`. Now, it does not take version as a parameter. Parses the `storageVersion` and updates it to fast, if needed, with the latest live tree version. Added more unit tests to make sure all code branches are covered